cpp/CMakeLists.txt

```diff
@@ -569,139 +153,207 @@ function(add_library_module NAMESPACE MODULE MODULE_SOURCE_DIRS)
         file(GLOB_RECURSE MODULE_SOURCE "${MODULE_SOURCE_DIR}/*")
         list(APPEND MODULE_SRC ${MODULE_SOURCE})
     endforeach(MODULE_SOURCE_DIR IN LISTS MODULE_SOURCE_DIRS)
-    message(STATUS "Adding library module with name ${MODULE} under namespace ${NAMESPACE} ...")
-    message(STATUS "Using sources: ${MODULE_SRC}")
+    message(STATUS "Adding library module ${NAMESPACE}_${MODULE} ...")
+    message(TRACE "Using sources: ${MODULE_SRC}")
     add_library("${NAMESPACE}_${MODULE}" "${MODULE_SRC}")
-    set_target_properties("${NAMESPACE}_${MODULE}" PROPERTIES BUILD_RPATH "\$ORIGIN")
+    set_target_properties("${NAMESPACE}_${MODULE}"
+        PROPERTIES BUILD_RPATH                         "\$ORIGIN"
+                   # set target compile options
+                   CXX_STANDARD                        14
+                   CXX_STANDARD_REQUIRED               ON
+                   CUDA_STANDARD                       14
+                   CUDA_STANDARD_REQUIRED              ON
+                   POSITION_INDEPENDENT_CODE           ON
+                   INTERFACE_POSITION_INDEPENDENT_CODE ON
+    )
+    target_compile_options("${NAMESPACE}_${MODULE}"
+                    PUBLIC "$<$<COMPILE_LANGUAGE:CXX>:${CUDF_CXX_FLAGS}>"
+                           "$<$<COMPILE_LANGUAGE:CUDA>:${CUDF_CUDA_FLAGS}>"
+    )
+    target_compile_definitions("${NAMESPACE}_${MODULE}"
+                    PUBLIC "$<$<COMPILE_LANGUAGE:CXX>:${CUDF_CXX_DEFINITIONS}>"
+                           "$<$<COMPILE_LANGUAGE:CUDA>:${CUDF_CUDA_DEFINITIONS}>"
+    )
     # Disable Jitify log printing. See https://github.com/NVIDIA/jitify/issues/79
     target_compile_definitions("${NAMESPACE}_${MODULE}" PRIVATE "JITIFY_PRINT_LOG=0")
-    # per-thread default stream
+    # Specify include paths for the current target and dependents
+    target_include_directories("${NAMESPACE}_${MODULE}"
+                   PUBLIC      "$<BUILD_INTERFACE:${DLPACK_INCLUDE_DIR}>"
+                               "$<BUILD_INTERFACE:${JITIFY_INCLUDE_DIR}>"
+                               "$<BUILD_INTERFACE:${LIBCUDACXX_INCLUDE_DIR}>"
```

Comment:
I'll have to double-check Jitify and libcu++, but DLPack is explicitly done this way because its include path might come from CPM, conda (via `find_path`), or overridden on the command line (for non-conda local builds) via `-DDLPACK_INCLUDE_DIR`: https://github.com/rapidsai/cudf/blob/fc85527c4fc04b590c91bf485823b4111054820f/cpp/cmake/thirdparty/CUDF_GetDLPack.cmake#L18-L36

Reply:
Chatted offline about this and the case this is trying to handle is when someone is developing on the dependency library alongside libcudf.